src/main/java/core/basesyntax/strategy/handler/SupplyOperationHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperationHandler implements OperationHandler {
+    private final FruitDao fruitDao;
+
+    public SupplyOperationHandler(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public void process(FruitTransaction transaction) {
+        Integer currentQuantity = fruitDao.get(transaction.getFruit());
```

Comment:
There's a potential null pointer exception if `fruitDao.get(transaction.getFruit())` returns null because the code then tries to unbox it to an int on line 20. Consider handling the case where `currentQuantity` is null.